libs/sdk-ui-pivot/src/impl/data/dataSourceUtils.ts

```diff
@@ -0,0 +1,61 @@
+// (C) 2019-2021 GoodData Corporation
+import { GridApi } from "@ag-grid-community/all-modules";
+import ApiWrapper from "../base/agGridApiWrapper";
+import isEqual from "lodash/isEqual";
+import { IGridTotalsRow } from "./resultTypes";
+import { ROW_SUBTOTAL, ROW_TOTAL } from "../base/constants";
+import { IDimension } from "@gooddata/sdk-model";
+
+export const areTotalsChanged = (gridApi: GridApi | undefined, newTotals: IGridTotalsRow[]): boolean => {
+    const currentTotalsCount = gridApi?.getPinnedBottomRowCount() ?? 0;
+    const newTotalsCount = newTotals === null ? 0 : newTotals.length;
```

Comment:
What if `newTotals === undefined`? Would `newTotals?.length ?? 0` be safer?

Reply:
fixed